src/details/ArborX_Callbacks.hpp

```diff
@@ -214,8 +214,8 @@ struct LegacyCallbackWrapper
   Callback _callback;
 
   template <typename Predicate>
-  KOKKOS_FUNCTION auto operator()(Predicate const &predicate,
-                                  Value const &value) const
+  KOKKOS_FUNCTION decltype(auto) operator()(Predicate const &predicate,
```

Comment:
Why would we want to change that?

Reply:
Good point. I'm not sure why I changed it.